### ISSUE TITLE
Pyodide doesn't play well with `vite build`

### ISSUE THREAD

**Issue by yongrenjie** [Urban-Analytics-Technology-Platform/demoland-web#67]
Build log from current dev branch (https://github.com/Urban-Analytics-Technology-Platform/demoland-web/commit/18fd07f3ac91f699c6102a2d811f78513d502307) attached below.

There's some discussion at #66 but I haven't quite gotten it to work on my machine yet.

```
$ npm run build

> demoland-web@2.0.0-alpha build
> vite build

vite v4.3.5 building for production...
transforming (104) node_modules/@kurkle/color/dist/color.esm.js[plugin:vite:resolve] Module "node:http" has been externalized for browser compatibility, imported by "/Users/jyong/demoland/web/web/node_modules/node-fetch/src/index.js". See http://vitejs.dev/guide/troubleshooting.html#module-externalized-for-browser-compatibility for more details.
[plugin:vite:resolve] Module "node:https" has been externalized for browser compatibility, imported by "/Users/jyong/demoland/web/web/node_modules/node-fetch/src/index.js". See http://vitejs.dev/guide/troubleshooting.html#module-externalized-for-browser-compatibility for more details.
[plugin:vite:resolve] Module "node:zlib" has been externalized for browser compatibility, imported by "/Users/jyong/demoland/web/web/node_modules/node-fetch/src/index.js". See http://vitejs.dev/guide/troubleshooting.html#module-externalized-for-browser-compatibility for more details.
[plugin:vite:resolve] Module "node:stream" has been externalized for browser compatibility, imported by "/Users/jyong/demoland/web/web/node_modules/node-fetch/src/index.js". See http://vitejs.dev/guide/troubleshooting.html#module-externalized-for-browser-compatibility for more details.
[plugin:vite:resolve] Module "node:buffer" has been externalized for browser compatibility, imported by "/Users/jyong/demoland/web/web/node_modules/node-fetch/src/index.js". See http://vitejs.dev/guide/troubleshooting.html#module-externalized-for-browser-compatibility for more details.
[plugin:vite:resolve] Module "node:stream" has been externalized for browser compatibility, imported by "/Users/jyong/demoland/web/web/node_modules/node-fetch/src/body.js". See http://vitejs.dev/guide/troubleshooting.html#module-externalized-for-browser-compatibility for more details.
[plugin:vite:resolve] Module "node:util" has been externalized for browser compatibility, imported by "/Users/jyong/demoland/web/web/node_modules/node-fetch/src/body.js". See http://vitejs.dev/guide/troubleshooting.html#module-externalized-for-browser-compatibility for more details.
[plugin:vite:resolve] Module "node:buffer" has been externalized for browser compatibility, imported by "/Users/jyong/demoland/web/web/node_modules/node-fetch/src/body.js". See http://vitejs.dev/guide/troubleshooting.html#module-externalized-for-browser-compatibility for more details.
[plugin:vite:resolve] Module "node:util" has been externalized for browser compatibility, imported by "/Users/jyong/demoland/web/web/node_modules/node-fetch/src/headers.js". See http://vitejs.dev/guide/troubleshooting.html#module-externalized-for-browser-compatibility for more details.
[plugin:vite:resolve] Module "node:http" has been externalized for browser compatibility, imported by "/Users/jyong/demoland/web/web/node_modules/node-fetch/src/headers.js". See http://vitejs.dev/guide/troubleshooting.html#module-externalized-for-browser-compatibility for more details.
[plugin:vite:resolve] Module "node:url" has been externalized for browser compatibility, imported by "/Users/jyong/demoland/web/web/node_modules/node-fetch/src/request.js". See http://vitejs.dev/guide/troubleshooting.html#module-externalized-for-browser-compatibility for more details.
[plugin:vite:resolve] Module "node:util" has been externalized for browser compatibility, imported by "/Users/jyong/demoland/web/web/node_modules/node-fetch/src/request.js". See http://vitejs.dev/guide/troubleshooting.html#module-externalized-for-browser-compatibility for more details.
[plugin:vite:resolve] Module "node:fs" has been externalized for browser compatibility, imported by "/Users/jyong/demoland/web/web/node_modules/fetch-blob/from.js". See http://vitejs.dev/guide/troubleshooting.html#module-externalized-for-browser-compatibility for more details.
[plugin:vite:resolve] Module "node:path" has been externalized for browser compatibility, imported by "/Users/jyong/demoland/web/web/node_modules/fetch-blob/from.js". See http://vitejs.dev/guide/troubleshooting.html#module-externalized-for-browser-compatibility for more details.
[plugin:vite:resolve] Module "node:net" has been externalized for browser compatibility, imported by "/Users/jyong/demoland/web/web/node_modules/node-fetch/src/utils/referrer.js". See http://vitejs.dev/guide/troubleshooting.html#module-externalized-for-browser-compatibility for more details.
[plugin:vite:resolve] Module "worker_threads" has been externalized for browser compatibility, imported by "/Users/jyong/demoland/web/web/node_modules/node-domexception/index.js". See http://vitejs.dev/guide/troubleshooting.html#module-externalized-for-browser-compatibility for more details.
[plugin:vite:resolve] Module "node:process" has been externalized for browser compatibility, imported by "/Users/jyong/demoland/web/web/node_modules/fetch-blob/streams.cjs". See http://vitejs.dev/guide/troubleshooting.html#module-externalized-for-browser-compatibility for more details.
[plugin:vite:resolve] Module "node:stream/web" has been externalized for browser compatibility, imported by "/Users/jyong/demoland/web/web/node_modules/fetch-blob/streams.cjs". See http://vitejs.dev/guide/troubleshooting.html#module-externalized-for-browser-compatibility for more details.
✓ 112 modules transformed.
✓ built in 2.56s
[vite:worker] "promisify" is not exported by "__vite-browser-external", imported by "node_modules/node-fetch/src/body.js".
file: /Users/jyong/demoland/web/web/src/lib/python/pyoide_worker.ts?worker:9:26
 7:
 8: import Stream, {PassThrough} from 'node:stream';
 9: import {types, deprecate, promisify} from 'node:util';
                              ^
10: import {Buffer} from 'node:buffer';
error during build:
RollupError: "promisify" is not exported by "__vite-browser-external", imported by "node_modules/node-fetch/src/body.js".
    at error (file:///Users/jyong/demoland/web/web/node_modules/rollup/dist/es/shared/node-entry.js:2125:30)
    at Module.error (file:///Users/jyong/demoland/web/web/node_modules/rollup/dist/es/shared/node-entry.js:13452:16)
    at Module.traceVariable (file:///Users/jyong/demoland/web/web/node_modules/rollup/dist/es/shared/node-entry.js:13863:29)
    at ModuleScope.findVariable (file:///Users/jyong/demoland/web/web/node_modules/rollup/dist/es/shared/node-entry.js:12418:39)
    at Identifier.bind (file:///Users/jyong/demoland/web/web/node_modules/rollup/dist/es/shared/node-entry.js:8116:40)
    at CallExpression.bind (file:///Users/jyong/demoland/web/web/node_modules/rollup/dist/es/shared/node-entry.js:5739:23)
    at CallExpression.bind (file:///Users/jyong/demoland/web/web/node_modules/rollup/dist/es/shared/node-entry.js:9669:15)
    at VariableDeclarator.bind (file:///Users/jyong/demoland/web/web/node_modules/rollup/dist/es/shared/node-entry.js:5739:23)
    at VariableDeclaration.bind (file:///Users/jyong/demoland/web/web/node_modules/rollup/dist/es/shared/node-entry.js:5735:28)
    at Program.bind (file:///Users/jyong/demoland/web/web/node_modules/rollup/dist/es/shared/node-entry.js:5735:28)
```

**Comment by yongrenjie** [comment]
Fixed by https://github.com/Urban-Analytics-Technology-Platform/demoland-web/commit/ebe0eed4d90ba75b6afe884fecb450af2e6f8957 and https://github.com/Urban-Analytics-Technology-Platform/demoland-web/commit/1007dde58a82753123da2d36491f286337819543

TL;DR: Solution was `npm i --save-dev isomorphic-fetch` and then add the following fields inside `vite.config.ts`

```typescript
export default defineConfig({
    ...,
    worker: { format: 'es' },
    resolve: {
        alias: {
            'node-fetch': 'isomorphic-fetch',
        },
    },
   ...
```

**Comment by yongrenjie** [comment]
Build works fine and preview works fine but only when deployed on port 5173. If viewing on port 4173 (e.g. with `npx vite preview`) this error happens:

```Cross-Origin Request Blocked: The Same Origin Policy disallows reading the remote resource at http://localhost:5173/data/median_form.parquet. (Reason: CORS request did not succeed). Status code: (null).```

**Comment by yongrenjie** [comment]
Closing as pyodide works very well now on the dev branch!
